web-marketplace/src/pages/CreateOrganization/CreateOrganization.tsx

```diff
@@ -0,0 +1,482 @@
+
+      if (organizationIdRef.current) {
+        console.info(
+          '[CreateOrganization] restored organization id from draft',
+          {
+            organizationId: organizationIdRef.current,
+          },
+        );
+      }
+
+      const existing = organizationProgress[storedDaoAddress];
+      const nextStep = Math.max(maxAllowedStep, 1);
+      if (!existing || existing.step !== nextStep || !existing.name) {
+        setOrganizationProgressStep(
+          storedDaoAddress,
+          nextStep,
+          storedName ?? organizationNameRef.current,
+        );
+      }
+      setHasUnfinishedOrganization(true);
+    }
+  }, [data, organizationProgress, maxAllowedStep]);
+
+  useEffect(() => {
+    const entries = Object.values(organizationProgress);
+    if (entries.length > 0) {
+      const entry = entries[0];
+      daoAddressRef.current = entry.daoAddress;
+      organizationNameRef.current = entry.name;
+      setHasUnfinishedOrganization(true);
+    } else {
+      daoAddressRef.current = undefined;
+      organizationNameRef.current = undefined;
+      setHasUnfinishedOrganization(false);
+    }
+  }, [organizationProgress]);
+
+  const currentProgressEntry = useMemo(
+    () => Object.values(organizationProgress)[0],
+    [organizationProgress],
+  );
+  const resumeStep = useMemo(() => {
+    if (!currentProgressEntry) return 0;
+    return Math.min(
+      currentProgressEntry.step ?? 0,
+      CREATE_ORG_STEPS.length - 1,
+    );
+  }, [currentProgressEntry]);
+
+  useEffect(() => {
+    if (resumeStep > 0 && activeStep === 0) {
+      handleActiveStep(resumeStep);
+    }
+  }, [resumeStep, activeStep, handleActiveStep]);
+
+  // No-op: modal starts open (showTransferModal=true) and will be closed via Skip/Yes handlers
+
+  const prefillFromActiveAccount = (): Partial<EditProfileFormSchemaType> => {
+    const name = activeAccount?.name || displayName;
+    const profileImage = activeAccount?.image || avatarUrl;
+    const backgroundImage = activeAccount?.bgImage || DEFAULT_PROFILE_BG;
+    const description = activeAccount?.description?.trimEnd() || '';
+    const websiteLink = activeAccount?.websiteLink || '';
+    const twitterLink = activeAccount?.twitterLink || '';
+    return {
+      name,
+      profileImage,
+      backgroundImage,
+      description,
+      websiteLink,
+      twitterLink,
+      // Ensure org type even if hidden
+      profileType: AccountType.Organization,
+    } as Partial<EditProfileFormSchemaType>;
+  };
+
+  const handleSkipTransfer = () => {
+    setShowTransferModal(false);
+  };
+
+  const handleYesTransfer = () => {
+    const values = prefillFromActiveAccount();
+    setOrgProfileInitialValues(values);
+    // persist values for this step so user can immediately proceed if desired
+    handleSave(values as Record<string, unknown>, 0);
+    setShowTransferModal(false);
+  };
+
+  const handleOrganizationProfileSaved = useCallback(
```

Comment:
this would make more sense to have this directly in OrganizationProfileStep code
imagine we put all the saving logic for all the steps in this file, it will become hard to read